pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -1597,6 +1602,21 @@ async def run(self):
                             if incorrect_changelogs or incorrect_modular_relation:
                                 log.debug("Updated data for package {}".format(result.nevra))
                                 await sync_to_async(result.save)()
+                            seen = []
+                            for path in d_content.content.files:
+                                if path in seen:
+                                    continue
+                                seen.append(path)
+                            if len(d_content.content.files) > len(seen):
+                                nevra = format_nevra(
+                                    name=d_content.content.name,
+                                    epoch=d_content.content.epoch,
+                                    version=d_content.content.version,
+                                    release=d_content.content.release,
+                                    arch=d_content.content.arch,
+                                )
+                                log.warn(f"Package {nevra} lists some files more than once")
```

Comment:
IMO you don't need to log it more than once, it will have already hit the first log statement at this point.